File: mcorg.backends/project-service/src/main/java/app/mcorg/organizer/presentation/rest/mappers/project/CreateProjectInputMapper.java
package app.mcorg.organizer.presentation.rest.mappers.project;

import app.mcorg.organizer.domain.usecase.project.CreateProjectUseCase;
import app.mcorg.organizer.presentation.rest.entities.project.CreateProjectRequest;
import lombok.experimental.UtilityClass;

@UtilityClass
public class CreateProjectInputMapper {
    public static CreateProjectUseCase.InputValues map(CreateProjectRequest request) {
        return new CreateProjectUseCase.InputValues(
                request.name()
        );
    }
}

File: webapp/src/main/kotlin/app/mcorg/presentation/mappers/project/CreateProjectInputMapper.kt
package app.mcorg.presentation.mappers.project

import app.mcorg.presentation.entities.project.CreateProjectRequest
import app.mcorg.presentation.mappers.*
import io.ktor.http.*

fun InputMappers.Companion.createProjectInputMapper(parameters: Parameters): CreateProjectRequest {
    val name = parameters.required("projectName")
    val dimension = EnumMappers.mapDimension(parameters.optional("dimension"))
    val priority = EnumMappers.mapPriority(parameters.optional("priority"))
    val requiresPerimeter = parameters.optionalBoolean("requiresPerimeter") ?: false

    return CreateProjectRequest(name, priority, dimension, requiresPerimeter)
}